js/stdlib/ts/ALGO.ts

```diff
@@ -1499,6 +1499,15 @@ export const connectAccount = async (networkAccount: NetworkAccount): Promise<Ac
       };
     };
 
+    const getCurrentStep_ = async (getC:GetC): Promise<BigNumber> => {
+      const { getAppState, getGlobalState } = await getC();
+      const appSt = await getAppState();
+      if ( !appSt ) { throw Error(`getCurrentStep_: no appSt`); }
+      const gs = await getGlobalState(appSt);
+      if ( !gs ) { throw Error(`getCurrentStep_: no gs`); }
+      return gs[0];
```

Comment:
I would look at what the view code does, because it already needs to learn this. At the very least you should make that code call this function